sdk/ml/azure-ai-ml/tests/pipeline_job/e2etests/test_pipeline_job.py

```diff
@@ -1344,6 +1344,133 @@ def test_pipeline_node_with_default_component(self, client: MLClient, randstr: C
+        register_node_output_path = "tests/test_configs/pipeline_jobs/helloworld_pipeline_job_register_node_output_without_name.yaml"
+        error_message_pipeline = "PipelineOutput.name is missing."
+        error_message_node = "NodeOutput.name is missing."
+
+        with pytest.raises(UserErrorException) as e:
+            pipeline = load_job(source=register_pipeline_output_path)
+            pipeline_job = client.jobs.create_or_update(pipeline)
+        assert error_message_pipeline in str(e.value)
+
+        with pytest.raises(UserErrorException) as e:
+            pipeline = load_job(source=register_node_output_path)
+            pipeline_job = client.jobs.create_or_update(pipeline)
+        assert error_message_node in str(e.value)
+
+    def test_register_output_without_name_sdk(self, client: MLClient):
+        from azure.ai.ml import dsl
+        component = load_component(source="./tests/test_configs/components/helloworld_component.yml")
+        component_input = Input(type='uri_file', path='https://dprepdata.blob.core.windows.net/demo/Titanic.csv')
+
+        @dsl.pipeline()
+        def register_node_output():
+            node = component(component_in_path=component_input)
+            node.outputs.component_out_path.version = '1'
+
+        with pytest.raises(UserErrorException) as e:
+            pipeline = register_node_output()
+            pipeline.settings.default_compute = "azureml:cpu-cluster"
+            pipeline_job = client.jobs.create_or_update(pipeline)
+        assert "NodeOutput.name is missing." in str(e.value)
+
+        @dsl.pipeline()
+        def register_pipeline_output():
+            node = component(component_in_path=component_input)
+            return {
+                'pipeine_a_output': node.outputs.component_out_path
+            }
+
+        with pytest.raises(JobParsingError) as e:
+            pipeline = register_pipeline_output()
+            pipeline.outputs.pipeine_a_output.version = '1'
+            pipeline.settings.default_compute = "azureml:cpu-cluster"
+            pipeline_job = client.jobs.create_or_update(pipeline)
+        assert "PipelineOutput.name is missing." in str(e.value)
+
+    def test_register_output_cli(self, client: MLClient):
+        register_pipeline_output_path = "./tests/test_configs/pipeline_jobs/helloworld_pipeline_job_register_pipeline_output_name_version.yaml"
+        pipeline = load_job(source=register_pipeline_output_path)
+        pipeline_job = client.jobs.create_or_update(pipeline)
+        output = pipeline_job.outputs.component_out_path
+        assert output.name == 'pipeline_output'
+        assert output.version == '1'
+
+        register_node_output_path = "./tests/test_configs/pipeline_jobs/helloworld_pipeline_job_register_node_output_name_version.yaml"
+        pipeline = load_job(source=register_node_output_path)
+        pipeline_job = client.jobs.create_or_update(pipeline)
```

Comment:
ensure we cancelled the created jobs

Reply:
replacded create_or_update() with assert_job_cancel()